packages/core/src/core/client.ts

```diff
@@ -164,8 +164,17 @@ export class GeminiClient {
     return this.getChat().getHistory();
   }
 
-  setHistory(history: Content[]) {
-    this.getChat().setHistory(history);
+  setHistory(history: Content[], stripThoughts = false) {
+    const historyToSet = stripThoughts
+      ? history.map((content) => ({
```

Comment:
can you clean this up a little bit? 
right now it is adding 
`thoughtSignature: undefined` 
properties everywhere. 
Would be nice to clone and then call 
delete part.thoughtSignature
if and only if part has a thoughtSignature field.
or something along those lines.
In addition, I think it is possible there are more deeply nested Parts containing other parts that might have thoughtSignature. Out of paranoia I would suggest creating a version of this that does a robust traversal of PartListUnion stripping thoughtSignature fields for all types that might have them.